test/e2e/node_reuse_test.go

```diff
@@ -0,0 +1,584 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	bmo "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	capm3 "github.com/metal3-io/cluster-api-provider-metal3/api/v1alpha5"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/util/taints"
+	"k8s.io/utils/pointer"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	kcp "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/test/framework"
+	"sigs.k8s.io/cluster-api/util/patch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	KUBERNETES_VERSION   = e2eConfig.GetVariable("KUBERNETES_VERSION")
+	UPGRADED_K8S_VERSION = "v1.21.3"
```

Comment:
As we might have different `UPGRADED_K8S_VERSION` version for node_reuse and upgrade test I am trying to keep it local as in the dev_env was declared in upgrade_vars or features_vars and not in common but probably later we can address this. Also I have got this issue while using the e2eConfig file https://jenkins.nordix.org/job/airship_metal3io_capi_m3_e2e_test_centos/104/console

Reply:
The error happenned when go packages were downloading. Hopefully it is one-time error. 
If you want to have different k8s versions for upgrade and node-reuse tests, you should have two different variables. Also, we should avoid to hardcode variables in the go code. It would be easier to update and maintain the test framework if we put these variables in the config file.

Reply:
I have fixed that I am trying to keep the same name like feature tests for easier reviews. I suggest once merging all the tests we can have follow up PR to refactor naming and structure in more independent and different way than Ansible